src/include/rbd/librbd.hpp

```diff
@@ -568,12 +570,15 @@ class CEPH_RBD_API Image
                             size_t info_size);
   int mirror_image_get_status(mirror_image_status_t *mirror_image_status,
 			      size_t status_size);
+  int mirror_image_get_service_id(std::string *service_id);
   int aio_mirror_image_promote(bool force, RBD::AioCompletion *c);
   int aio_mirror_image_demote(RBD::AioCompletion *c);
   int aio_mirror_image_get_info(mirror_image_info_t *mirror_image_info,
                                 size_t info_size, RBD::AioCompletion *c);
   int aio_mirror_image_get_status(mirror_image_status_t *mirror_image_status,
                                   size_t status_size, RBD::AioCompletion *c);
+  int aio_mirror_image_get_service_id(std::string *service_id,
```

Comment:
Nit: doesn't seem to be needed/used right now?

Reply:
Yes. I was using this initially for `rbd mirror pool status`, in `StatusImageRequest` until I realized that I could just call `mirror_image_service_id_list` once and pass the list to `StatusImageRequest`.
Ok. I will remove those.